lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -159,6 +161,106 @@ static bool isZeroBitType(Type type) {
   return false;
 }
 
+// structs and logic/reg/wire are base types that arrays are built up from
+// and they are printed independently of their context.  Integers and arrays and
+// unpacked arrays create non-trivial array bound definitions on each side of
+// the name.  So we break printing up into those three parts: baseType
+// (struct,logic,reg,wire), packed dimensions (integer bitwidth, packed array
+// width), name (left to caller), and unpacked dimensions.
+
+// Print out the packed array subscripts and integer bitwidth after a wire/port
+// declaration.
+static void printArraySubscriptsPre(Type type, Location loc, raw_ostream &os) {
+  TypeSwitch<Type, void>(type)
+      .Case<InOutType>([&](InOutType inout) {
+        printArraySubscriptsPre(inout.getElementType(), loc, os);
+      })
+      .Case<ArrayType>([&](ArrayType array) {
+        os << '[' << (array.getSize() - 1) << ":0]";
+        printArraySubscriptsPre(array.getElementType(), loc, os);
+      })
+      .Case<UnpackedArrayType>([&](UnpackedArrayType array) {
+        printArraySubscriptsPre(array.getElementType(), loc, os);
+      })
+      .Case<IntegerType>([&](IntegerType integer) {
+        if (integer.getWidth() == 0)
+          os << "/*Zero Width*/";
+        else if (integer.getWidth() != 1)
+          os << '[' << (integer.getWidth() - 1) << ":0]";
+      })
+      .Case<StructType>([&](StructType) {})
+      .Default([&](Type t) {
+        mlir::emitError(loc, "value has an unsupported verilog type ") << t;
+      });
+}
+
+// Print out the unpacked array subscripts after a variable/port/field name.
+static bool hasArraySubscriptsPre(Type type) {
```

Comment:
Why not combine this with `printArraySubscriptsPre`? Also, the code doc is wrong.